codelabs/codelab-02/codelab2README.md

```diff
@@ -0,0 +1,501 @@
+
+![Admin Permissions](../../../media/codelabs/codelab-02/admin-group.png)
+
+3. Click through to the "Complete" page where AWS will provide you with the access key ID and secret access key. Keep the latter safe, perhaps saving them in a text file on your machine, because AWS will not show it to you again. These keys are used to sign programmatic requests that you make to AWS.
+
+##### Testing the CLI
+
+If you haven't already done so, set up your local development environment. Follow the instructions here: [Environment Setup](https://github.com/UMD-CS-STICs/389Lspring18/blob/master/env.md). Upon entering the final command, you will be prompted to give the keys you just obtained. Set us-east-1 and json as the default region and output respectively.
+
+$ aws configure 
+AWS Access Key ID [None]: AKIAIOSFODNN7EXAMPLE 
+AWS Secret Access Key [None]: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY 
+Default region name [None]: us-east-1 
+Default output format [None]: json
+
+Now test your credentials with the following command.
+
+$ aws its get-caller-identity
+
+Good to go. We'll use the CLI more below.
+
+##### Root User
+
+To clarify the difference between the user you have been using to log in to the AWS Console and the IAM user you just created:
+
+The credentials you've been using to log in with are the root user credentials. Normally, you avoid using the root account and instead federate access out to IAM users so that you can audit their usage. However, in this class, we will continue to use the root account for simplicity.
+
+Essentially, you have two users for your account: the root user, which you use to access the AWS console, and the IAM user, which you use to programmatically access AWS. You won't need to think about this now that your AWS environment is configured, but it is good to know.
+
+### S3 Basics
+
+There are two main ways that you will programmatically interact with AWS. The first is through the CLI and the second is through the boto3 SDK. The CLI is great for one-time changes to your AWS environment. However, if you want to create dynamic scripts, then you will want to use something else — in this class, that will be Python with Amazon's Python SDK, boto3.
+
+For codelabs, I'll be showing you how to interact with AWS via the CLI, since the CLI is easier to experiment with compared to Python code. I won't go into as much detail with the boto3 library, since I don't believe that explaining both adds much value. The APIs for both are fairly similar, so you should be able to easily pick up the relevant boto3 as long as you understand each codelab.
+
+Feel free to ask on Piazza if you have any trouble or questions about boto3 APIs.
+
+#### References
+
+You want to reference the [aws s3api](http://docs.aws.amazon.com/cli/latest/reference/s3api/index.html) documentation while working through this codelab.
```

Comment:
Bold: "Reference the `aws s3api` documentation while working on the codelab" etc.